rust_src/src/alloc.rs

```diff
@@ -57,4 +65,39 @@ pub fn bool_vector(args: &mut [LispObject]) -> LispObject {
     vector
 }
 
+/// Create a new record.
+/// TYPE is its type as returned by `type-of'; it should be either a
+/// symbol or a type descriptor.  SLOTS is the number of non-type slots,
+/// each initialized to INIT.
+#[lisp_fn]
+pub fn make_record(r#type: LispObject, slots: EmacsUint, init: LispObject) -> LispObject {
+    let size = slots + 1;
+    unsafe {
+        let ptr = allocate_record(size as i64);
+        let contents = (*ptr).contents.as_mut_slice(size as usize);
+        contents[0] = r#type;
+        for rec in contents.iter_mut().skip(1) {
+            *rec = init;
+        }
+        make_lisp_ptr(ptr as *mut c_void, Lisp_Vectorlike)
```

Comment:
AFAIK we use tag_ptr in rust

Reply:
This only works on `ExternalPtr`s though, it seems unnecessary to create an `ExternalPtr` only to immediately extract its contained pointer.